simplepdf.py

```diff
@@ -2,104 +2,85 @@
+        N = include page N;
+        -1 = last page;
+        Example: 1:5,~3,7,-1 will include pages 1,2,4,5,7 and the last page
+"""
 
 def main():
-    commands = {
-        "--merge": lambda opts: merge(parse_merge_args(opts)),
-        "--extract": lambda opts: extract(*parse_extract_args(opts)),
-        "--rotate": lambda opts: rotate(*parse_rotate_args(opts)),
-        "--insert": lambda opts: insert(*parse_insert_args(opts))
-    }
+    parser = argparse.ArgumentParser(description='Simple program for manipulating pdfs')
 
-    if len(sys.argv) < 2 or sys.argv[1] not in commands:
-        exit_showing_help()
+    subparsers = parser.add_subparsers(dest='subparser_name')
 
-    command = sys.argv[1]
-    opts = sys.argv[2:]
-    commands[command](opts)
+    parser_merge = subparsers.add_parser('merge', help='Merge pdfs into one')
+    parser_merge.add_argument('input_pdfs', nargs='*', help='List of pdf files to merge')
+    parser_merge.add_argument('output_pdf', help='Merged pdf file name')
+    parser_merge.set_defaults(func=merge)
 
+    parser_extract = subparsers.add_parser('extract', help='Extract pages from pdf')
+    parser_extract.add_argument('input_pdf', help='pdf to be extracted')
+    parser_extract.add_argument('page_string', help=PAGE_STRING_HELP)
+    parser_extract.add_argument('output_pdf', help='Extracted pdf file name')
+    parser_extract.set_defaults(func=extract)
 
-def exit_showing_help():
-    help()
-    sys.exit(1)
+    parser_rotate = subparsers.add_parser('rotate', help='Rotate given pages of a pdf with a given angle')
+    parser_rotate.add_argument('input_pdf', help='pdf to be rotated')
+    parser_rotate.add_argument('page_string', help=PAGE_STRING_HELP)
+    parser_rotate.add_argument('angle', type=int, help='Angle to rotate clockwise in degrees')
+    parser_rotate.add_argument('output_pdf', help='Rotated pdf file name')
+    parser_rotate.set_defaults(func=rotate)
 
+    parser_insert = subparsers.add_parser('insert', help='Insert a pdf inside another pdf at a given position')
+    parser_insert.add_argument('input_pdf', help='Base pdf file')
+    parser_insert.add_argument('inserted_pdf', help='pdf to be inserted into input_pdf')
+    parser_insert.add_argument('position', type=int, help='Page where the pdf should be inserted')
+    parser_insert.add_argument('output_pdf', help='Final pdf file name')
+    parser_insert.set_defaults(func=insert)
 
-def help():
-    print("simplepdf [--merge] [in1.pdf ... inN.pdf] [out.pdf]")
-    print("simplepdf [--rotate] [in.pdf] [pages] [angle_degree (CW)] [out.pdf]")
-    print("simplepdf [--extract] [in.pdf] [pages] [out.pdf]")
-    print("simplepdf [--insert] [in.pdf] [add.pdf] [page] [out.pdf]\n")
-    print(
-        "Page selection: \n   1:N = from 1 to N (inclusive)\n\
-    ~N = exclude page N\n\
-    N = include page N\n\
-    -1 = last page\n\
-    Example: 1:5,~3,7,-1 will include pages 1,2,4,5,7 and the last page"
-    )
+    args = parser.parse_args()
```

Comment:
Calling simplepdf.py without arguments raises "AttributeError: 'function' object has no attribute 'func'". This could be fixed including -h when no argument is passed.

```suggestion
    args = parser.parse_args(args=None if sys.argv[1:] else ['-h'])
```